mri/scripts/gridsearch.py

```diff
@@ -0,0 +1,289 @@
+        self.regularizer_op = regularizer_op_class(**regularizer_op_kwargs)
+        self.optimizer_kwargs = optimizer_kwargs
+
+    def reconstruct_case(self, kspace_data, fourier_op,
+                         reconstructor_class, reconstructor_kwargs,
+                         fourier_params=None):
+        """Internal Function to carry out reconstruction for a
+        special case. This function pulls in appropriate keyword arguments
+        from input and declares appropriate Linear, Fourier and Regularizer
+        Operators. These operators later are used to create the image model
+        by defining the reconstructor. Then the reconstruction is carried and
+        results are returned
+
+        Parameters
+        ----------
+        kspace_data: np.ndarray
+            the kspace data for reconstruction
+        fourier_op: object of class FFT
+            this defines the fourier operator. for NonCartesianFFT, please make
+            fourier_op as `None` and pass fourier_params to allow
+            parallel execution
+        reconstructor_class: class
+            reconstructor class
+        reconstructor_kwargs: dict
+            extra kwargs for reconstructor
+        fourier_params: dict, default None
+            holds dictionary with init_class pointing to fourier
+            class to be used and args having keyword arguments for
+            initialization
+            This is passed only if fourier_op is None so that fourier_op can be
+            made on spot during reconstruction.
+            NOTE: We declare fourier operator inside this function to allow
+            parallel execution as NonCartesianFFT cannot be pickled.
+        """
+        if fourier_op is None:
+            fourier_op = fourier_params['init_class'](
+                **fourier_params['kwargs']
+            )
+        reconstructor = reconstructor_class(
+            fourier_op=fourier_op,
+            linear_op=self.linear_op,
+            regularizer_op=self.regularizer_op,
+            **reconstructor_kwargs,
+        )
+        raw_results = reconstructor.reconstruct(
+            kspace_data=kspace_data,
+            **self.optimizer_kwargs
+        )
+        return raw_results
+
+
+def gather_result(metric, results, metric_direction=None):
+    """ Gather the best reconstruction result.
+
+    Parameters:
+    -----------
+    metric: str,
+        the name of the metric, it will become a dict key in the output dict.
+    metric_direction: bool,
+        if True the higher the better the metric value is (like for `ssim`),
+        else the lower the better.
+    results: list of list,
+        list of the raw results of the gridsearch
+
+    Return:
+    -------
+    results and location of best results in given set of raw results
+    """
+    # Make metric string lower case
+    metric = metric.lower()
+    list_metric = np.array([
+        res[2][metric]['values'][-1] for res in results
+    ])
+    if metric_direction is None:
```

Comment:
add a comment saying what's happening underneath.

Also you could add `'mse'` as a supported metric.